recipe/meta.yaml

```diff
@@ -1,13 +1,13 @@
 {% set name = "ml_dtypes" %}
-{% set version = "0.4.0" %}
+{% set version = "0.4.1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/ml_dtypes-{{ version }}.tar.gz
-  sha256: eaf197e72f4f7176a19fe3cb8b61846b38c6757607e7bf9cd4b1d84cd3e74deb
+  sha256: fad5f2de464fd09127e49b7fd1252b9006fb43d2edc1ff112d390c324af5ca7a
 
 build:
   script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

Comment:
The run dependency should be `numpy >1.20` instead of using pin_compatible.

[jax-ml/ml_dtypes@`v0.4.0...v0.4.1`#diff-50c86b7ed8](https://github.com/jax-ml/ml_dtypes/compare/v0.4.0...v0.4.1#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R20)

Reply:
Thanks I added:
```
    - numpy >1.20     # [py<310]
    - numpy >=1.21.2  # [py>=310]
    - numpy >=1.23.3  # [py>=311]
    - numpy >=1.26.0  # [py>=312]
```

To match upstream: https://github.com/jax-ml/ml_dtypes/blob/v0.4.1/pyproject.toml#L20-L23